library/src/test/java/com/novoda/storagepathfinder/AndroidDeviceStorageInspectorTest.java

```diff
@@ -0,0 +1,143 @@
+package com.novoda.storagepathfinder;
+
+import android.content.Context;
+
+import java.io.File;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+
+public class AndroidDeviceStorageInspectorTest {
+
+    private static final StoragePath.Type PRIMARY = StoragePath.Type.PRIMARY;
+    private static final StoragePath.Type SECONDARY = StoragePath.Type.SECONDARY;
+
+    private static final StoragePath PRIMARY_BASE_PATH = DeviceStoragePath.create("/primary-base-path", PRIMARY);
+    private static final StoragePath PRIMARY_APPLICATION_PATH = DeviceStoragePath.create("/primary-application-path", PRIMARY);
+
+    private static final StoragePath SECONDARY_BASE_PATH_1 = DeviceStoragePath.create("/secondary-base-path-1", SECONDARY);
+    private static final StoragePath SECONDARY_BASE_PATH_2 = DeviceStoragePath.create("/secondary-base-path-2", SECONDARY);
+    private static final StoragePath SECONDARY_APPLICATION_PATH_1 = DeviceStoragePath.create("/secondary-application-path-1", SECONDARY);
+
+    private Context context = mock(Context.class);
+    private AndroidSystem androidSystem = mock(AndroidSystem.class);
+    private FileSystem filesystem = mock(FileSystem.class);
+    private DeviceFeatures devicesFeatures = mock(DeviceFeatures.class);
+
+    private CommonDirectories commonDirectories = new CommonDirectories() {
+        @Override
+        public File getExternalStorageDirectoryBasePath() {
+            return PRIMARY_BASE_PATH.asFile();
+        }
+
+        @Override
+        public File getExternalStorageDirectoryApplicationPath() {
+            return PRIMARY_APPLICATION_PATH.asFile();
+        }
+    };
+
+    private DeviceStorageInspector storageInspector;
+
+    @Before
+    public void setup() {
+        given(filesystem.getCommonDirectories()).willReturn(commonDirectories);
+        given(androidSystem.getEnv(anyString())).willReturn(SECONDARY_BASE_PATH_1.asString());
+
+        storageInspector = new AndroidDeviceStorageInspector(context, filesystem, devicesFeatures, androidSystem);
+    }
+
+    @Test
+    public void canGetThePrimaryStorageBasePath() {
+        StoragePath basePath = storageInspector.getPrimaryStorageBasePath();
+
+        assertThat(basePath).isEqualTo(PRIMARY_BASE_PATH);
+    }
+
+    @Test
+    public void canGetThePrimaryStorageApplicationPath() {
+        StoragePath basePath = storageInspector.getPrimaryStorageApplicationPath();
+
+        assertThat(basePath).isEqualTo(PRIMARY_APPLICATION_PATH);
+    }
+
+    @Test
+    public void canGetASecondaryStorageBasePathFromTheExternalDirectoryInspector() {
```

Comment:
This and the tests below seem to be leaking the implementation, I don't know anything about these inspectors from this classes point of view. Should we be writing these as a blackbox?